team/bundles/org.eclipse.compare/compare/org/eclipse/compare/LabelContributionItem.java

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) ETAS GmbH 2023, all rights reserved.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ETAS GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.compare;
+
+import org.eclipse.jface.action.ControlContribution;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Label;
+
+
+/**
+ * @since 3.10
+ *
+ *        A contribution item which delegates to a label on the tool bar.
+ *
+ */
+public class LabelContributionItem extends ControlContribution {
+
+	private Label toolbarLabel;
+	private String labelName;
+
+	/**
+	 * @param id
+	 * @param name
+	 */
+	public LabelContributionItem(String id, String name) {
+		super(id);
+		this.labelName = name;
+	}
+
+	@Override
+	protected Control createControl(Composite parent) {
+		Composite composite = new Composite(parent, SWT.LEFT);
```

Comment:
Few issues here:
1) `SWT.LEFT` used in constructors of COmposite & Label  don't make any sense as the argument is not alignment => `SWT.NONE`
2) hard coded margins looks strange => remove
3) I would not align label on bottom, it looks odd as it is not aligned with buttons anymore => `SWT.CENTER`

With that, please try
```
	@Override
	protected Control createControl(Composite parent) {
		Composite composite = new Composite(parent, SWT.NONE);
		this.toolbarLabel = new Label(composite, SWT.NONE);

		GridLayout compositeLayout = new GridLayout(1, false);
		composite.setLayout(compositeLayout);
		GridData labelLayout = new GridData(SWT.LEFT, SWT.CENTER, true, true);

		this.toolbarLabel.setLayoutData(labelLayout);
		this.toolbarLabel.setText(this.labelName);
		return composite;
	}
```

Reply:
> 2\. hard coded margins looks strange => remove

If we remove these the label looks like below (Not aligned properly)

![diffLabel](https://github.com/eclipse-platform/eclipse.platform/assets/130035155/1ede8eb7-6f74-4a1c-861d-fa0e8459e80d)

Reply:
> > 2. hard coded margins looks strange => remove
> 
> If we remove these the label looks like below (Not aligned properly)
> 
> ![diffLabel](https://private-user-images.githubusercontent.com/130035155/287603452-1ede8eb7-6f74-4a1c-861d-fa0e8459e80d.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE2NzAwNjYsIm5iZiI6MTcwMTY2OTc2NiwicGF0aCI6Ii8xMzAwMzUxNTUvMjg3NjAzNDUyLTFlZGU4ZWI3LTZmNzQtNGExYy04NjFkLWZhMGU4NDU5ZTgwZC5KUEc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwNFQwNjAyNDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNjQ2OTU4Yjk1YjRkMGI0NWNhNmUzZWFkYjNmNjI5NWQ1NzhlODA3NTA4ODFiZGVhNGYwYjQzMWU0ZTE5ZjAzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.cwJWykgsJoXXuNFxlw_WiRkXdVA2TgBnY73wwKSoYsQ)

Which OS is it?

Reply:
In that case I would only enable Windows specific workaroumds in "if(isWindows)" blocks.

Reply:
> In that case I would only enable Windows specific workaroumds in "if(isWindows)" blocks.

Have you verified on Linux and Mac already if this is working without margin values ?

Reply:
I've verified on Linux, I have no Mac access.

Reply:
> 3\. I would not align label on bottom, it looks odd as it is not aligned with buttons anymore => `SWT.CENTER`

Here , I observed that label appearance does not change whatever SWT value I use (tested in Ubuntu and Windows). Where can we see the impact ?

Reply:
> In that case I would only enable Windows specific workaroumds in "if(isWindows)" blocks.

I believe this "special handling of margins on Windows" should be better also commented in-line in the code, as it will most probably confuse someone sooner or later. Anyway, thank you both for a good job on this PR! 👍